### PR TITLE
INT-2545 Add support for customizing startup

### PR DESCRIPTION
**Pull request by adrianpowell**
Jira: https://issues.sonatype.org/browse/INT-2545

Adds support for users to customize the startup in order to delete lock files or perform other checks. Also added customization for the grace period.

### CODE REVIEW

File: OpenShift/helm/nexus-iq/LICENSE [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Copyright (c) 2016-present Sonatype, Inc.
```
Comment (adrianpowell) [inline OpenShift/helm/nexus-iq/LICENSE]: boyscoutting - required for cert

File: OpenShift/helm/nexus-repository-manager/LICENSE [hunk 2]
```diff
@@ -1,86 +1,13 @@
-Eclipse Public License - v 1.0
-THE ACCOMPANYING PROGRAM IS PROVIDED UNDER THE TERMS OF THIS ECLIPSE PUBLIC LICENSE ("AGREEMENT"). ANY USE, REPRODUCTION OR DISTRIBUTION OF THE PROGRAM CONSTITUTES RECIPIENT'S ACCEPTANCE OF THIS AGREEMENT.
+Copyright (c) 2016-present Sonatype, Inc.
```
Comment (adrianpowell) [inline OpenShift/helm/nexus-repository-manager/LICENSE]: boyscoutting - used the wrong license

File: OpenShift/helm/nexus-iq/values.yaml [hunk 3]
```diff
@@ -36,6 +36,15 @@ serviceAccount:
   # If not set and create is true, a name is generated using the fullname template
   name:
 
+deployment:
+  # Uncomment and modify this to run a command before starting the IQ Server
+  preStart:
+    command:    # '["/bin/rm", "-f", "/path/to/lockfile"]'
```
Comment (jflinchbaugh) [inline OpenShift/helm/nexus-iq/values.yaml]: I forget what we did with the other one. Can we get concrete paths to the lock file, as that might be the most common case? Or do we not yet want to encourage this behavior yet?

Comment (adrianpowell) [reply OpenShift/helm/nexus-iq/values.yaml]: The path to the logfile is version-dependent and complicated. It's certainly not obvious at a glance what it is or why it's there. Not to mention it's something I don't want to try to maintain. I would rather provide an example that is clear to read even if it is not immediately usable.

**Pull request merged**